### PR TITLE
update tree-sitter-markdown

### LINKED ISSUE #6762: Markdown inline-code highlighting fails if opening backtick is followed by dollar sign (conflict with inline-LaTeX?)
### Working: The following text renders correctly, with both instances of `variable` rendered in inline-code color:

```
Text `variable` text `variable`
```

#### GitHub rendering:
Text `variable` text `variable`

#### Helix rendering:
image

### Broken: Starting with a dollar sign after the variable makes the second "text" get incorrectly colored in inline-code color

```
Text `$variable` text `variable`
```

#### GitHub rendering:
Text `$variable` text `variable`

#### Helix rendering:
image

The first backtick appears to get ignored.

## Reproduction Steps

 

I tried this:

1. `hx`
2. `:lang markdown`
3. Copy above markdown snippet
4. `<space> p`

## I expected this to happen:

Helix rendering `$variable` in gray and `text` in white.

### Instead, this happened:

Helix rendering `$variable` in white and `text` in gray; 
the next following inline-code snippet in the same paragraph also becomes white (default text color) again.

All highlighting is shifted by one backtick, as the backtick-dollar combination appears to not be recognized.

This might be caused by a conflict with inline-LaTeX.

Notably, trying to escape `$` with `\` does not help:
image

## Helix log

No relevant log output.

## Platform

openSUSE Tumbleweed - Linux

## Terminal Emulator

Konsole

## Helix Version

helix master bb275421997b9e2bd63b04227b7e345a193818d5

### PR DESCRIPTION
**Pull request by king-11**
Fixes #6762 

Signed-off-by: Lakshya Singh <lakshay.singh1108@gmail.com>

### CODE REVIEW

**Comment by king-11** [comment]
@the-mikedavis requesting your review

**Comment by LeoniePhiline** [comment]
Upstream grammar changes:

 https://github.com/MDeiml/tree-sitter-markdown/compare/7e7aa9a25ca9729db9fe22912f8f47bdb403a979...fa6bfd51727e4bef99f7eec5f43947f73d64ea7d#diff-09e9c3bd5118602f994bcc272a683e0de2002d95343505f0b0fac0e89c7c67c2

File: languages.toml [hunk 1]
```diff
@@ -1126,7 +1126,7 @@ indent = { tab-width = 2, unit = "  " }
 
 [[grammar]]
 name = "markdown"
-source = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "7e7aa9a25ca9729db9fe22912f8f47bdb403a979", subpath = "tree-sitter-markdown" }
+source = { git = "https://github.com/MDeiml/tree-sitter-markdown", rev = "fa6bfd51727e4bef99f7eec5f43947f73d64ea7d", subpath = "tree-sitter-markdown" }
```
[the remark this answers is not in the document]
Comment (king-11) [reply languages.toml:1129]: Sure

**Comment by king-11** [comment]
Updated revision for both

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**